eZ/Publish/Core/Persistence/Legacy/URL/Query/CriterionHandler/MatchNone.php

```diff
@@ -24,8 +24,11 @@ public function accept(Criterion $criterion)
     /**
      * {@inheritdoc}
      */
-    public function handle(CriteriaConverter $converter, SelectQuery $query, Criterion $criterion)
-    {
-        return $query->expr->not($query->bindValue('1'));
+    public function handle(
+        CriteriaConverter $converter,
+        QueryBuilder $queryBuilder,
+        Criterion $criterion
+    ) {
+        return $queryBuilder->expr()->eq(1, 0);
```

Comment:
IMHO should be consistent with 
https://github.com/ezsystems/ezplatform-kernel/pull/22/files#diff-b0c101048d5d5b566ea06b14e2d72d23R32 or other way around.

```php
return '1 == 0''
```

Reply:
Hmm, ok, let's go with string. Fixed via ec0ac44. Though it's difficult to guess which direction Doctrine will go. Right now it's `CompositeExpression|string`, but mostly `string` :/